### ISSUE TITLE
Tabela de atividades e contribuições do usuário

### ISSUE THREAD

**Issue by marlonangeli** [filipedeschamps/tabnews.com.br#1599]
### Descrição

Fala turma! Após ver algumas issues como a #1576 e a rede de qualificação, surgiu a ideia de criar algo semelhante à tabela de contribuição do GitHub.

![tabela de contribuição](https://github.com/filipedeschamps/tabnews.com.br/assets/73134662/88a05116-e473-4322-8ab5-a6347f0a824c)

Adicionar a cada dia um somatório de publicações, comentários e TabCoins/TabCash ganhos. Seria legal para analisar a progressão dos usuários na plataforma.

### Sugestão de implementação

Quantificar os posts e comentários não seria um problema, quanto aos TabCoins e TabCash a discussão em #1341 já mostra uma análise utilizando datas e que poderia ser útil.

**Comment by mthmcalixto** [comment]
@marlonangeli Talvez seja possível, só precisava saber como retornar ou gravar esses dados.

Tem repositório aqui no Github que você pode colocar os dados e ele iria retornar tudo bonitinho, se chama [React Activity Calendar](https://github.com/grubersjoe/react-activity-calendar).

Pelo que entendi basta tem dados:
```json
[
  {
    "date": "2023-06-14",
    "count": 2,
    "level": 1
  },
  {
    "date": "2023-06-22",
    "count": 16,
    "level": 3
  }
]
```

Se tivesse de alguma forma conseguir retornar à quantidade de cada ação realizada no dia, era só juntar esses dados e contar.

**Comment by Rafatcb** [comment]
É uma sugestão interessante. Tem um issue com uma sugestão para termos um extrato dos TabCoins: https://github.com/filipedeschamps/tabnews.com.br/issues/1341. Acredito que se optarmos pela forma consolidada, facilitará as buscas no banco para realizarmos esse tipo de gráfico.

Vale lembrar que no GitHub, ao clicar em um quadrado, é possível ver os detalhes do dia. Aí já entra a preocupação em definir o que deve ser privado e o que pode ser público. Talvez nada deva ser público, mas podemos debater melhor isso.

**Comment by mthmcalixto** [comment]
@marlonangeli coloquei embaixo do perfil e ficou bem interessante, @Rafatcb acredito que detalhar o que foi feito no dia não seja tão importante assim, e sobre os ganhos de TabsCoins, acho que não precisa ser colocado como "atividade";

![Screenshot_30](https://github.com/filipedeschamps/tabnews.com.br/assets/121470728/b60a99eb-edf9-4b6b-afe9-4ce3855ea498)
![Screenshot_31](https://github.com/filipedeschamps/tabnews.com.br/assets/121470728/ad882e33-199c-48c3-9750-ce5d0e56529b)

**Comment by aprendendofelipe** [comment]
Essa ideia é ótima candidata para ser desenvolvida e disponibilizada de forma totalmente independente, apenas utilizando dados da API pública.

Algo como o https://github.com/anuraghazra/github-readme-stats

**Comment by mthmcalixto** [comment]
> Essa ideia é ótima candidata para ser desenvolvida e disponibilizada de forma totalmente independente, apenas utilizando dados da API pública.
> 
> Algo como o https://github.com/anuraghazra/github-readme-stats

@aprendendofelipe sim, realmente iria fazer mais sentindo, apenas passando o **username** do usuário do TabNews.

O mesmo autor do [React Activity Calendar](https://github.com/grubersjoe/react-activity-calendar) fez um API que retorna todos as atividades de um **username** do GitHub, [github-contributions-api](https://github.com/grubersjoe/github-contributions-api), talvez poderia criar algo semelhante na API do TabNews e então deixar que usem de forma independente o [React Activity Calendar](https://github.com/grubersjoe/react-activity-calendar).
